src/js/media-chrome-button.ts

```diff
@@ -95,19 +95,25 @@ template.innerHTML = /*html*/ `
 class MediaChromeButton extends globalThis.HTMLElement {
```

Comment:
I migrated this one component as part of proving out/validating any issues.

Reply:
The issue is that when the utility functions that except `HTMLElement`s are converted to TS, they still don't consider these true `HTMLElement`s:

![Screenshot 2024-06-17 at 16 56 34](https://github.com/muxinc/media-chrome/assets/1508707/a321fea0-7998-4f5c-9156-6b59986d4f8d)

Reply:
Do we know of a specific environment/configuration that breaks without `server-safe-globals.js`? I have a few ideas for workarounds, but I have no way to verify them without a test case.

Reply:
yep, a Next.js app should reproduce the issues that come up without patching the globals.
an example here https://github.com/muxinc/media-chrome/tree/main/examples/nextjs-with-typescript

Reply:
The next's doesn't need the globals at all:

https://github.com/littlespex/media-chrome/pull/4